File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> product(n+1, 1), ans(n);
        for (int i = 0; i < n; i++) {
            product[i+1] = product[i] * nums[i];
        }
        int d=1;
        for (int i = n-1; i>=0; i--) {
            ans[i] = product[i] * d;
            d *= nums[i];
        }
        return ans;
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        ans = []
        curr = []
        def backtrack(i: int, c_sum: int):
            nonlocal ans, nums, target, curr
            if c_sum > target: return
            if c_sum == target:
                ans.append(curr.copy())
                return
            
            for idx in range(i, len(nums)):
                curr.append(nums[idx])
                backtrack(idx, nums[idx] + c_sum)
                curr.pop()
        backtrack(0, 0)
        return ans